### PR TITLE
[MRG]Update notebook examples

### LINKED ISSUE #294: Update stale  Jupyter Notebook examples
#### Software and Hardware Versions 
ModelCI v1.x.x, CUDA Version vx.x.x, GPU device used...

#### Problem description

Currently the jupyter notebook examples are out-of-date, thus need to be updated 

#### Other Information 
Things you tried, stack traces, related issues, suggestions on how to fix it...

### PR DESCRIPTION
**Pull request by univerone**
Closes #294

#### Before submitting this PR, please make sure you have:

- [x] run `python -m pytest tests/` and confirmed all checks still pass.
- [x] verified that any code or assets from external sources are properly credited.

### CODE REVIEW

File: modelci/app/readme.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 ## Start server
 
 ```shell script
-uvicorn main:app
+uvicorn modelci.app.main:app --port 8000
```
Comment (YuanmingLeee) [inline modelci/app/readme.md:6]: Shall we assume user start the application from the current directory

File: modelci/hub/manager.py [hunk 2]
```diff
@@ -101,8 +99,7 @@ def register_model(
         }
 
         for model in models:
-            model.model_status = [ModelStatus.PROFILING]
-            ModelService.update_model(model)
+            update_model(str(model.id), ModelUpdateSchema(model_status=[ModelStatus.PROFILING]))
```
Comment (YuanmingLeee) [inline modelci/hub/manager.py:102]: we should persist existing model status, and add a `PROFILING` tag to it.

File: modelci/hub/manager.py [hunk 3]
```diff
@@ -218,8 +217,9 @@ def get_remote_model_weight(model: ModelBO):
     save_path.parent.mkdir(exist_ok=True, parents=True)
 
     if not save_path.exists():
+        # TODO save TFS or TRT model files from gridfs
         with open(str(save_path), 'wb') as f:
-            f.write(model.weight.weight)
+            f.write(model.weight.__bytes__())
```
Comment (YuanmingLeee) [inline modelci/hub/manager.py:222]: You should use `bytes(model.weight)` instead.

File: modelci/hub/manager.py [hunk 4]
```diff
@@ -195,21 +193,22 @@ def generate_model_family(
...
-        converter.convert(net, 'tfs', 'trt', tf_path=tfs_dir, save_path=trt_dir, inputs=inputs, outputs=outputs, max_batch_size=32)
+        converter.convert(net, 'tfs', 'trt', tf_path=tfs_dir, save_path=trt_dir, inputs=inputs, outputs=outputs,
+                          max_batch_size=32)
         generated_dir_list.append(trt_dir.with_suffix('.zip'))
 
     return generated_dir_list
 
 
-def get_remote_model_weight(model: ModelBO):
+def get_remote_model_weight(model: MLModel):
```
Comment (YuanmingLeee) [inline modelci/hub/manager.py:203]: Is it better to put all local cache management to a separate file, e.g., `cache_manager`?

File: modelci/persistence/service_.py [hunk 5]
```diff
@@ -53,23 +55,53 @@ def save(model_in: MLModel):
...
     # TODO: update weight ID in the MLModelIn
     weight_id = _fs.put(bytes(model_in.weight), filename=model_in.weight.filename)
-    model = MLModel(**model_in.dict(exclude={'weight'}), weight=weight_id)
+    model = MLModel(**model_in.dict(exclude={'weight', 'model_input'}), weight=weight_id)
     model.id = _collection.insert_one(model.dict(exclude_none=True, by_alias=True, use_enum_values=True)).inserted_id
     return model
 
 
-def get_by_id(id: str) -> MLModel:
+def get_by_id(_id: str) -> MLModel:
```
Comment (YuanmingLeee) [inline modelci/persistence/service_.py:63]: use `id_` instead of `_id`. Otherwise, your model argument will be protected.

**Review by YuanmingLeee — COMMENTED** [review]
LGTM, is this WIP?

File: modelci/app/readme.md (hunk 1 above)
Comment (univerone) [reply modelci/app/readme.md:6]: maybe we should use modelci CLI instead

Comment (YuanmingLeee) [reply modelci/app/readme.md:6]: I don't think so, this backend app is independent of CLI.

File: modelci/app/readme.md [hunk 6]
```diff
@@ -3,7 +3,7 @@
 ## Start server
 
 ```shell script
-uvicorn main:app
+uvicorn main:app --port 8000
```
Comment (YuanmingLeee) [inline modelci/app/readme.md:6]: Default is 8000?

File: modelci/hub/manager.py [hunk 7]
```diff
@@ -12,146 +12,61 @@
...
     """Query a model by name, task, framework, engine or version.
 
     Arguments:
-        architecture_name (str): Model architecture name.
+        architecture (str): Model architecture name.
         task (Task): which machine learn task is model used for,Default to None
         framework (Framework): Framework name, optional query key. Default to None.
         engine (Engine): Model optimization engine name.
-        version (ModelVersion): Model version. Default to None.
+        version (Int): Model version. Default to None.
```
Comment (YuanmingLeee) [inline modelci/hub/manager.py:39]: `int`

File: modelci/hub/cache_manager.py [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+        ) for pair
+        in pairs
+    ]
+
+    # get weights of newest version of each pair
+    for model_group in model_groups:
+        get_remote_model_weight(model_group[0])
+
+
+def delete_remote_weight(model: MLModel):
```
Comment (YuanmingLeee) [inline modelci/hub/cache_manager.py:70]: delete_cache_weight

File: modelci/hub/registrar.py [hunk 9]
```diff
@@ -141,8 +140,9 @@ def register_model(
         }
 
         for model in models:
-            model.model_status = [ModelStatus.PROFILING]
-            ModelService.update_model(model)
+            if ModelStatus.PROFILING not in model.model_status:
+                model.model_status.append(ModelStatus.PROFILING)
```
Comment (YuanmingLeee) [inline modelci/hub/registrar.py:144]: This tag should be added in the profiler module.

File: modelci/types/models/mlmodel.py [hunk 10]
```diff
@@ -48,9 +48,9 @@ def __init__(self, __root__):
 
     @property
     def filename(self):
-        if self.file:
+        if hasattr(self, 'file') and self.file is not None:
```
Comment (YuanmingLeee) [inline modelci/types/models/mlmodel.py:51]: Use `getattr(self, 'file', None)`

File: tests/test_modelhub_cli.py [hunk 11]
```diff
@@ -11,20 +12,24 @@
 runner = CliRunner()
 file_dir = f"{str(Path.home())}/.modelci/ResNet50/pytorch-pytorch/image_classification"
```
Comment (YuanmingLeee) [inline tests/test_modelhub_cli.py:13]: str(Path.home() / '.modelci/ResNet50/pytorh-pytorch/image_classification')

File: tests/test_modelhub_cli.py [hunk 12]
```diff
@@ -53,20 +58,18 @@ def test_update():
...
+
+
+def test_delete():
+    with requests.get(f'{app_settings.api_v1_prefix}/model/') as r:
+        model_list = r.json()
+    for model in model_list:
+        model_id = model["id"]
+        result = runner.invoke(app, ['delete', model_id])
+        assert result.exit_code == 0
+        assert f"Model {model_id} deleted\n" == result.output
```
Comment (YuanmingLeee) [inline tests/test_modelhub_cli.py:75]: why the output is a string?

File: modelci/app/readme.md (hunk 6 above)
Comment (univerone) [reply modelci/app/readme.md:6]: Yes, according to https://www.uvicorn.org/settings/#socket-binding, default port is 8000.

File: tests/test_modelhub_cli.py (hunk 12 above)
Comment (univerone) [reply tests/test_modelhub_cli.py:75]: Here the `result` variable is a `click.testing.Result` instance, its `output` property will return a string which represent the standard output, refer to https://click.palletsprojects.com/en/8.0.x/api/#click.testing.Result.output

**Comment by HuaizhengZhang** [comment]
Seems this is not only improving notebook

Please detail the information about the other changes.

@univerone

**Comment by univerone** [comment]
> Seems this is not only improving notebook
> 
> Please detail the information about the other changes.
> 
> @univerone

I'll try to split this pr to make it more clear

**Pull request closed without merging**